ingestor/adx/tasks.go

```diff
@@ -134,58 +134,100 @@ func (t *SyncFunctionsTask) Run(ctx context.Context) error {
 		return fmt.Errorf("failed to list functions: %w", err)
 	}
 	for _, function := range functions {
-
-		if function.Spec.Database != v1.AllDatabases && function.Spec.Database != t.kustoCli.Database() {
+		availableDB := t.kustoCli.Database()
+		configuredDB := function.Spec.Database
+		if configuredDB != v1.AllDatabases && !strings.EqualFold(configuredDB, availableDB) {
```

Comment:
Is there a way we can set these conditions (database mismatch, criteria mismatch) once instead of n-times for the number of Kusto dbs we have? I think every worker for each database will repeatedly set these every reconcile period for ones that don't have matching configs.